### DISCUSSION TITLE
What does MuseScore 4.x lack compared to other so-called more professional music notation software?

### DISCUSSION THREAD

**Discussion by jeremy-donson** [musescore/MuseScore#11542] [Q&A]
This discussion is about specific v 4.x feature shoot-outs for open source authoring, publishing and collaboration.
We can keep the [3.x discussion](https://musescore.org/en/node/315333) going on the web site and start the 4.x discussion here.

This would also be a good place for advertising both free and paid musescore web services, as well as the non-OSS mobile apps.
This way we can strategize with regard to fostering adoption by legacy users of finalé, sibelius, etc.

The open source contributions (many kinds) are among the best features to include (where others have nada).

- Good use for a github page.
- Also a good place for supporting file migrations from those tools.
- Also a good place for supporting MuseScore file upgrades from 3.x to 4.x. (Do it to a copy or 👎🏼 !! )

**Comment by jeremy-donson** [comment]
https://pages.github.com/ => Best way to do docs with git workflow....

https://gists.github.com => Gists are the easy way to learn GHMD (github flavored markdown)

A shootout github page doc for features would be excellent on a github page.

A list of finale and sibelius version features would be nice:  https://www.youtube.com/watch?v=bWOnApl8TF4

**Comment by jeremy-donson** [comment]
[Finale uses SmartMusic service](https://michaelgood.info/)

[MusicXML Spec Repo](https://github.com/w3c/musicxml)

[2018: Feature Shootout](https://musescore.org/en/user/199521/blog/2018/03/12/comparison-between-musescore-2-finale-v25-dorico-scorecloud-studio-and)

**Comment by brendencallahan** [comment]


**Reply by MarcSabatella** [reply]
Actually, in current versions of MuseScore (eg, 3.6.2), you *can* assign shortcuts to the accidental commands.  Just search for "note input" to find them in the list.  Probably you looked for "flat" etc, but those words don't appear in the command names - the accidental characters themselves do.  Also, even without the direct accidentals, the usual way people enter accidentals via shortcut is Up/Down, although this won't help for courtesy accidentals.

For MU4, there will be default shortcuts for the accidental commands themselves..

**Reply by brendencallahan** [reply]
Thank you for the reply. My comment was in regards to the current build of MU4, as there are not default shortcuts and no way to add them. It's nice to know they will be supported/already are in current versions.

**Reply by MarcSabatella** [reply]
Indeed, it's a known bug that the command with no shortcuts already defined are missing from the list of customizable shortcuts in current pre-release MU4 builds.  They are actually there; the dialog is just not showing them.  That will presumably be fixed before release.

**Comment by worldwideweary** [comment]
Here are a few things that come to mind:

1. Flat slur options
2. Cross-staff/voice arpeggio with appropriate playback by default
3. Beams connected between systems
4. Vertical stacked chord symbol extensions by default options instead of xml editing required ...and maybe easier polychord stacking (although those are kind of rare it seems)
5. User customized options for slur position defaults... Also taking into better account fingering positions
6. Maybe better laissez vibre options by default
7. Load custom SMuFL fonts externally and also allow for per-symbol replacements
8. Good midi CC control (hope for 4.x to have...)
9. The binding between external video and\or audio sources like with online services would be nice to have locally with time stretch markers
10. More Annotation abilities out of the box
11. Beyond twelve tone standard tuning playback in an easy way
12. Some fun things like vertical instrument names .... Hand writing recognition for a stylus-style input.
13. A user friendly tutoring of the sequencer so that correct MIDI input is necessary to move the playhead forward for Midi instrument training. Lots of paid for online services do this. If MS did it for free (i have a personal build for 3.6x that does it... Not that difficult to do) it would be cool since MS is now into training videos on the .com side...

**Reply by DaffodilAura** [reply]
Agreed for number 11 specifically- I use 159tet often.

**Comment by riaanvn** [comment]
I am not trying to answer the question, but trying to frame it in a way that encourages better participation. Would it be possible to change the subject from
… other so-called more professional music notation software?
to
… commercial music notation software?

(I had to drop the “other” since keeping it would imply that MuseScore is commercial. The change also implies that MuseScore’s only real competition is commercial, proprietary software, which I believe it is, but am not 100% certain)

Some the 3.x thread on the forums had similar issues with calling MuseScore not or less professional.
There is a tendency to conflate open source and freely distributable with non-professional, and closed source / proprietary / commercial with professional, which I believe the “so-called less professional” label perpetuates.

**Reply by MarcSabatella** [reply]
LilyPond is the main competitor that is both "professional" and free and open source.  It's a pretty different animal, though.

**Comment by Skelbti** [comment]
Could the functionality to import audio directly into the score as a wavesform like in StaffPad be added to MuseScore one day?

**Comment by papasax** [comment]
As a big band arranger, I really miss the scrub function from finale:
https://www.finalemusic.com/blog/scrubbing-your-music/

Lots of users miss it a lot:
https://musescore.org/en/node/291097#comment-1150929

**Reply by EricFulHat** [reply]
Thank you for bringing this up - seems like people have been asking for a while. Seems like a small feature to implement that would have a big impact for many people, myself included. The looping playback feature is not really the same.

**Comment by Kqpout** [comment]
Coming from Finale, the one feature I miss is the ability to easily and rapidly move one bar, or a selection of bars,  up on the previous staff before the line break. You just select the measure(s), click up and it's done. 
I know how to do it in MS but it's a pain in the ass, and sometimes I want to force it even if the bars are already at minimum spacing and stretching.
I understands that the up and down keys have another function in MS, and a useful one, but I'd love a way to move these bars more easily.

**Comment by EricFulHat** [comment]
I switched from finale to musescore and now proselytize the program. The only thing I miss is the **_scrubbing playback function_** - this is VERY conducive to the compositional process. While I compose at the piano, the scrubbing function essentially enables me to 'fiddle around' with things and hear them back instantaneously when I'm away from the keyboard - **it is essentially a near substitute for sitting with the instrument and facilitating the compositional process.** When I am traveling, I compose on finale and import to Musescore because of this.

**Comment by johann-petrak** [comment]
Extremely disappointed after I tried to make it work with my MP11SE with MIDI over USB, I discovered how lacking proper MIDI support is in MuseScore 4 - basically does not work at all. 

This is much better in MuseScore 3, though even there it seems that not everything works with MIDI over USB (and MuseScore 4 on Windows 10, with Linux this seems to be even harder) as expected. 

I think both support and documentation of MIDI features need a lot of improvement. 
bility.

**Comment by DanielGHaas** [comment]
I am in the process of switching from Finale to MuseScore and this is the feature my workflow misses the most: Being able to specify chords with a base note using either "/" or "_" to have them displayed side-by-side or vertically with a horizontal line between them. Only offering the "/" option doesn't give the option for more compact and easily-readable chords when dealing with complex chords and many chord changes per measure. I understand the purists' argument about poly chords, but in practice, the two are used interchangeably to fit the style and spacing of the score. The fact that MusicXML supports three different arrangements supports the argument for providing the functionality.

**Comment by Fourniture** [comment]
I would say 1) dedicated formatting options like vertically numbered staff-sharing instruments, 2) chant notation, 3) compatability with Noteperformer.
Those are just some of the main ones, though. I have heard some people wanting various types of modern (read: unconventional) notation (though still staff-based), and I myself would appreciate more professional looking parts, but there are a lot more pressing issues...

**Comment by Rusca8** [comment]
Keyboard shortcuts (or, the option to assign them) for dynamics.

**Comment by BenignPigeon** [comment]
1.  Being able to paste time signatures
2.  Support for film scoring (video playing in app)

**Reply by AlexelTech** [reply]
Yes, synchronized video playback for scoring is really missing !
I know it's a big thing to implement, obviously, but it's keeping a lot of people back about switching from Dorico or Finale to Musescore...

**Reply by MarcSabatella** [reply]
> Yes, synchronized video playback for scoring is really missing ! I know it's a big thing to implement, obviously, but it's keeping a lot of people back about switching from Dorico or Finale to Musescore...

Download the “VLC Sync” plugin from Musescore.org, it allows MuseScore to synchronize with free & open source VLC Media Player.

**Comment by Rusca8** [comment]
Finale has `Alt+click` for pasting the currently selected bars to the cursor position (without ever needing to copy them).
It's really fast and lets you keep other things on the clipboard while you move things around.

...and Musescore doesn't seem to have any behavior for `Alt+click` so it should be clean to implement it.

**Reply by jeetee** [reply]
Update to at least 4.5 and Alt/Option+drag to copy a selection is right there.

**Reply by Rusca8** [reply]
Yes but no. They added an option to drag them, but dragging is quite cumbersome, specially if you want to copy things to places that are far away from the original selected music.

What I was saying is in Finale you **don't need to drag**. You can just Alt+Click on a bar, and it'll paste the selection (as if you had dragged it there). Slightly different workflow (quite more streamlined, imo).

**Reply by jeetee** [reply]
Aha! You can probably tell I've not used Finale much in my life :)

Sounds like a good feature request to request then!

**Reply by Rusca8** [reply]
Yeah, it's one of the things I'm missing after switching over to musescore. 😌
